respy/simulate.py

```diff
@@ -664,7 +638,7 @@ def _process_input_df_for_simulation(df, method, options, optim_paras):
 
     other_periods = df.query("period != 0")
     has_nans = np.any(other_periods.drop(columns="type", errors="ignore").isna())
-    if has_nans:
+    if has_nans and method == "one_step_ahead":
```

Comment:
Beyond the first period? I would assume that this is rather difficult because
- the imputed values need to be consistent with the later history.
- the imputation model becomes more and more complex for later periods because you could condition on the history.
Or do you have a simpler solution in mind?

Another thing I just recognized is that we should not impute values in the first period for the one-step-ahead simulation because of consistency. We might sample 7 years of schooling, but the value in the second period is 9 which is inconsistent. Would you agree?

Reply:
I have been thinking about a situation with a very few nans but I think you are right! 
I  am not sure whether I get your point about the one step ahead simulation. But I agree that it would be more consistent to not replace these since we dont do it from the second period onwards and would not be any different I think.

Reply:
Assume one individual has missing years of schooling and choice edu in the first period and 9 years of schooling in the second period. The value of years of schooling in the first period has to be 8, but sampling might return a different value. In this case, the user has to fix the data.